lib/mojito/request.ex

```diff
@@ -54,4 +54,12 @@ defmodule Mojito.Request do
   def validate_request(_request) do
     {:error, %Error{message: "request must be a map"}}
   end
+
+  @doc ~S"""
+  Converts non-string header values to UTF-8 string if possible.
+  """
+  @spec convert_headers_values_to_string(Mojito.request()) :: {:ok, Mojito.request()}
+  def convert_headers_values_to_string(%__MODULE__{headers: headers} = request) do
+    {:ok, %__MODULE__{request | headers: Headers.convert_values_to_string(headers)}}
```

Comment:
The `__MODULE__` here is not necessary.

Reply:
You mean better to use `%Request{...}`?

Reply:
You can just use `%{request | ...}` -- the resulting map or struct will always match what `request` is.

Reply:
Done